instrumentation/apache-pulsar/apache-pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/telemetry/PulsarSingletons.java

```diff
@@ -29,101 +32,104 @@ public final class PulsarSingletons {
 
-  private static final SpanNameExtractor<Message<?>> CONSUMER_RECEIVE =
-      new MessagingSpanNameExtractor<>(SpanKind.CONSUMER, MessageOperation.RECEIVE);
-  private static final SpanNameExtractor<Message<?>> CONSUMER_PROCESS =
-      new MessagingSpanNameExtractor<>(SpanKind.CONSUMER, MessageOperation.PROCESS);
-  private static final SpanNameExtractor<Message<?>> PRODUCER_SEND =
-      new MessagingSpanNameExtractor<>(SpanKind.PRODUCER, MessageOperation.SEND);
-
-  private static final Instrumenter<Message<?>, Attributes> CONSUMER_LISTENER_INSTRUMENTER =
-      createConsumerListenerInstrumenter();
-  private static final Instrumenter<Message<?>, Attributes> CONSUMER_RECEIVE_INSTRUMENTER =
+  private static final Instrumenter<PulsarRequest, Void> CONSUMER_PROCESS_INSTRUMENTER =
+      createConsumerProcessInstrumenter();
+  private static final Instrumenter<PulsarRequest, Void> CONSUMER_RECEIVE_INSTRUMENTER =
       createConsumerReceiveInstrumenter();
-  private static final Instrumenter<Message<?>, Attributes> PRODUER_INSTRUMENTER =
+  private static final Instrumenter<PulsarRequest, Void> PRODUER_INSTRUMENTER =
       createProducerInstrumenter();
 
-  public static Instrumenter<Message<?>, Attributes> consumerListenerInstrumenter() {
-    return CONSUMER_LISTENER_INSTRUMENTER;
+  public static Instrumenter<PulsarRequest, Void> consumerProcessInstrumenter() {
+    return CONSUMER_PROCESS_INSTRUMENTER;
   }
 
-  public static Instrumenter<Message<?>, Attributes> consumerReceiveInstrumenter() {
+  public static Instrumenter<PulsarRequest, Void> consumerReceiveInstrumenter() {
     return CONSUMER_RECEIVE_INSTRUMENTER;
   }
 
-  public static Instrumenter<Message<?>, Attributes> producerInstrumenter() {
+  public static Instrumenter<PulsarRequest, Void> producerInstrumenter() {
     return PRODUER_INSTRUMENTER;
   }
 
-  private static Instrumenter<Message<?>, Attributes> createConsumerReceiveInstrumenter() {
-    MessagingAttributesGetter<Message<?>, Attributes> getter =
+  private static Instrumenter<PulsarRequest, Void> createConsumerReceiveInstrumenter() {
+    MessagingAttributesGetter<PulsarRequest, Void> getter =
         PulsarMessagingAttributesGetter.INSTANCE;
 
-    return Instrumenter.<Message<?>, Attributes>builder(
-            TELEMETRY, INSTRUMENTATION_NAME, CONSUMER_RECEIVE)
-        .addAttributesExtractor(ConsumerAttributesExtractor.INSTANCE)
+    return Instrumenter.<PulsarRequest, Void>builder(
+            TELEMETRY,
+            INSTRUMENTATION_NAME,
+            MessagingSpanNameExtractor.create(getter, MessageOperation.RECEIVE))
+        .addAttributesExtractor(createMessagingAttributesExtractor(MessageOperation.RECEIVE))
         .addAttributesExtractor(
-            MessagingAttributesExtractor.create(getter, MessageOperation.RECEIVE))
+            NetClientAttributesExtractor.create(new PulsarNetClientAttributesGetter()))
         .buildConsumerInstrumenter(MessageTextMapGetter.INSTANCE);
```

Comment:
This should probably use the common `messagingReceiveInstrumentationEnabled` flag
```suggestion
        .setEnabled(ExperimentalConfig.get().messagingReceiveInstrumentationEnabled())
        .buildConsumerInstrumenter(MessageTextMapGetter.INSTANCE);
```